IGNITE-25466 .NET: Update NuGet README with platform compute info (#5889)
apache/ignite-3: modules/platforms/dotnet/README.md

## Compute

Compute API is used to execute distributed computations on the cluster. 

Compute jobs can be implemented in Java or .NET. 
Resulting binaries (jar or dll files) should be [deployed](https://ignite.apache.org/docs/ignite3/latest/developers-guide/code-deployment/code-deployment) to the server nodes and called by the full class name.

> Java job invocation example:

Reviewer: This would read better as a subsection. Maybe something like this
```suggestion
### Call a Java Compute Job

You can call [Java computing jobs](https://ignite.apache.org/docs/ignite3/latest/developers-guide/compute/compute) from your .NET code, for example:
```

```csharp
IList<IClusterNode> nodes = await client.GetClusterNodesAsync();
IJobTarget<IEnumerable<IClusterNode>> jobTarget = JobTarget.AnyNode(nodes);

var jobDesc = new JobDescriptor<string, string>(
    JobClassName: "org.foo.bar.MyJob",
    DeploymentUnits: [new DeploymentUnit("Unit1")]);

IJobExecution<string> jobExecution = await client.Compute.SubmitAsync(
    jobTarget, jobDesc, "Job Arg");

string jobResult = await jobExecution.GetResultAsync();
```

### Implement a .NET Compute Job

> 1. Prepare a "class library" project for the job implementations (`dotnet new classlib`). Prefer a separate project for compute jobs to reduce deployment size.

Reviewer: ```suggestion
1. Prepare a "class library" project for the job implementation (`dotnet new classlib`). In most cases, it is better to use a separate project for compute jobs to reduce deployment size.
```

2. Add a reference to `Apache.Ignite` package to the class library project (`dotnet add package Apache.Ignite`).

> 3. Create a class that implements `IComputeJob<TArg, TRes>` interface

Reviewer: ```suggestion
3. Create a class that implements `IComputeJob<TArg, TRes>` interface, for example:
```

    ```csharp
    public class HelloJob : IComputeJob<string, string>
    {
        public ValueTask<string> ExecuteAsync(IJobExecutionContext context, string arg, CancellationToken cancellationToken) =>
            ValueTask.FromResult("Hello " + arg);
    }
    ```

> 4. Publish the project (`dotnet publish -c Release`)

Reviewer: ```suggestion
4. Publish the project (`dotnet publish -c Release`).
```

5. Copy the resulting dll file (and any extra dependencies, EXCLUDING Ignite dlls) to a separate directory.

>    * Note: The directory must not contain any subdirectories.

Reviewer: ```suggestion
   * Note: The directory with the dll must not contain any subdirectories.
```

by specifying the assembly-qualified class name and using the `JobExecutorType.DotNetSidecar` option.

```csharp
var jobTarget = JobTarget.AnyNode(await client.GetClusterNodesAsync());
var jobDesc = new JobDescriptor<string, string>(
    JobClassName: typeof(HelloJob).AssemblyQualifiedName!,
    DeploymentUnits: [new DeploymentUnit("unit1")],
    Options: new JobExecutionOptions(ExecutorType: JobExecutorType.DotNetSidecar));

IJobExecution<string> jobExec = await client.Compute.SubmitAsync(jobTarget, jobDesc, "world");
```

Alternatively, use `JobDescriptor.Of` shortcut method to create a job descriptor from a job instance:

```csharp
JobDescriptor<string, string> jobDesc = JobDescriptor.Of(new HelloJob())
    with { DeploymentUnits = [new DeploymentUnit("unit1")] };
```

#### .NET Compute Server-side Requirements

* .NET compute jobs are executed in a separate process (sidecar) on the server node. 
* The process is started on the first .NET job call and then reused for subsequent jobs.
* Every deployment unit combination is loaded into a separate [AssemblyLoadContext](https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/understanding-assemblyloadcontext).

> * .NET 8 (or later) runtime (not SDK) is required on the server node.

Author: Agree, fixed.
